pkgs/development/python-modules/Theano/default.nix

```diff
@@ -0,0 +1,87 @@
+{ stdenv
+, runCommandCC
+, fetchFromGitHub
+, buildPythonPackage
+, pythonOlder
+, numpy
+, six
+, scipy
+, nose
+, nose-parameterized
+, pydot_ng
+, sphinx
+, pygments
+, future
+, cudaSupport ? false
+, cudatoolkit ? null
+, pycuda ? null
+, libgpuarray ? null
+, cudnn ? null
+}:
+
+assert cudaSupport
+  -> cudatoolkit != null
+  && libgpuarray != null
+  && pycuda != null
+  && cudnn != null;
+
+buildPythonPackage rec {
+  name = "Theano-cuda-${version}";
+  version = "0.8.2";
+
+  src = fetchFromGitHub {
+    owner = "Theano";
+    repo = "Theano";
+    rev = "46fbfeb628220b5e42bf8277a5955c52d153e874";
+    sha256 = "1sl91gli3jaw5gpjqqab4fiq4x6282spqciaid1s65pjsf3k55sc";
+  };
+
+  doCheck = false;
+
+  patches = [ ./compiler.patch ];
+  postPatch =
+    let
+      # Closure for running a command in a stdenv for building with given buildInputs
+      wrapped = command: buildInputs:
```

Comment:
The issue I ran into is that Theano tries to compile the code it generates with just a standard g++ call, and then expects to be able to dynamically load the results into itself.  Generally this goes wrong because

1. the system g++ cannot find the required libraries or, even if the first succeeds,

2. the resulting binary is not compatible with running Theano (i.e., depends on the system versions of common libraries like glibc which, especially if you are not using NixOS, may not match the ones Theano was compiled with).

Ideally I think you would want the runtime builds to execute in the same sort of environment that build-time builds were done in.  Hence the above code to capture an appropriate stdenv.  I [emailed the nix-dev mailing list](https://mailman.science.uu.nl/pipermail/nix-dev/2017-June/023893.html) about this earlier, but haven't received any replies.

It would be possible to further patch Theano or use `makeWrapper`/`wrapProgram` to add the appropriate compiler flags so it can find its libraries.  I think the above may be more elegant and less maintenance though as you automatically get the full build environment (i.e., the closure of the required includes, the setting of the appropriate NIX_* flags to enforce compiler purity, etc.).